bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/datalayer/DataLayerUtils.java

```diff
@@ -0,0 +1,148 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2020 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.models.datalayer;
+
+import com.adobe.cq.wcm.core.components.internal.DataLayerConfig;
+import com.adobe.cq.wcm.core.components.internal.Utils;
+import com.adobe.cq.wcm.core.components.models.Component;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+import com.day.cq.wcm.api.Template;
+import com.day.cq.wcm.api.components.ComponentContext;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.caconfig.ConfigurationBuilder;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Optional;
+
+import static com.adobe.cq.wcm.core.components.internal.Utils.ID_SEPARATOR;
+
+/**
+ * Utility helper functions for Data Layer.
+ */
+public final class DataLayerUtils {
```

Comment:
I see your point in creating a utility class that can be useful when creating custom components (https://github.com/adobe/aem-core-wcm-components/pull/1115#discussion_r499668347).
As the methods that relate to ID generation can be used in context that are not necessarily specific to the data layer (like HTML ID), I would move this class into `com.adobe.cq.wcm.core.components.models` and rename it `Utils`.

Reply:
I would rather we put the utility method in something like `com.adobe.cq.wcm.core.components.util.Component`. We could have more utility methods and classes in the future, seems wise to separate them.

Reply:
how about `com.adobe.cq.wcm.core.components.util.ComponentUtils`, which makes it clear what it is when using it? E.g.: `String id = ComponentUtils.generateId(...)`

Reply:
I renamed it to it as `com.adobe.cq.wcm.core.components.util.ComponentUtils` as suggested by @jckautzmann;
I prefer having `Utils` in the name because it is unambiguous that this is a utility class, and naming the class `Component` would cause a lot of class name conflicts, as this utility is likely to be used in conjunction with the `Component` model.  

If you convinced that this is the appropriate solution (as I am), then please feel free to resolve the conversation.